File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/presentation/api/v1/model/AccountResponseData.java
package kookmin.software.capstone2023.timebank.presentation.api.v1.model;

import kookmin.software.capstone2023.timebank.domain.model.Account;
import kookmin.software.capstone2023.timebank.domain.model.AccountType;

public class AccountResponseData {

    private final Long id;
    private final AccountType type;
    private final String name;
    private final ProfileResponseData profile;

    public AccountResponseData(Long id, AccountType type, String name, ProfileResponseData profile) {
        this.id = id;
        this.type = type;
        this.name = name;
        this.profile = profile;
    }

    public Long getId() {
        return id;
    }

    public AccountType getType() {
        return type;
    }

    public String getName() {
        return name;
    }

    public ProfileResponseData getProfile() {
        return profile;
    }

    public static class ProfileResponseData {
        private final String nickname;
        private final String imageUrl;

        public ProfileResponseData(String nickname, String imageUrl) {
            this.nickname = nickname;
            this.imageUrl = imageUrl;
        }

        public String getNickname() {
            return nickname;
        }

        public String getImageUrl() {
            return imageUrl;
        }
    }

    public static AccountResponseData fromDomain(Account account) {
        return new AccountResponseData(
                account.getId(),
                account.getType(),
                account.getName(),
                account.getProfile() != null ? new ProfileResponseData(
                        account.getProfile().getNickname(),
                        account.getProfile().getImageUrl()
                ) : null
        );
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/presentation/api/v1/model/AccountResponseData.kt
package kookmin.software.capstone2023.timebank.presentation.api.v1.model

import kookmin.software.capstone2023.timebank.domain.model.Account
import kookmin.software.capstone2023.timebank.domain.model.AccountType

data class AccountResponseData(
    val id: Long,
    val type: AccountType,
    val name: String,
    val profile: ProfileResponseData?,
) {
    data class ProfileResponseData(
        val nickname: String,
        val imageUrl: String,
    )

    companion object {
        fun fromDomain(account: Account): AccountResponseData {
            return AccountResponseData(
                id = account.id,
                type = account.type,
                name = account.name,
                profile = account.profile?.let {
                    ProfileResponseData(
                        nickname = it.nickname,
                        imageUrl = it.imageUrl,
                    )
                },
            )
        }
    }
}
